gnomad/utils/constraint.py

```diff
@@ -0,0 +1,103 @@
+# noqa: D100
+
+from typing import Union
+
+import hail as hl
+
+
+def annotate_mutation_type(
+    t: Union[hl.MatrixTable, hl.Table], trimer: bool = False
+) -> Union[hl.MatrixTable, hl.Table]:
+    """
+    Annotate variant types.
+
+    The following annotations are added to the output Table:
+        - cpg
+        - transition
+        - variant_type
+        - variant_type_model
+
+    :param t: Input Table or MatrixTable.
+    :param trimer: Whether to use trimers for context. Defaults to False (uses heptamers as context).
```

Comment:
If trimer is set to true here, but the context is actually formatted as a heptamer within the ht/mt this function wouldn't behave as expected. I think it's actually safer to drop this parameter and within the function set the mid_index by first checking the length of context to see if it is formatted as a trimer or a heptamer.